hoot-core/src/main/cpp/hoot/core/conflate/polygon/MultiPolygonCreator.cpp

```diff
@@ -377,10 +377,14 @@ deque<ConstWayPtr> MultiPolygonCreator::_orderWaysForRing(const vector<ConstWayP
     if (w->getNodeId(0) == lastId)
     {
       result.push_back(w);
+      //  Update the last id with the new last id
+      lastId = w->getLastNodeId();
```

Comment:
If you look in line 361 and 362 you'll see that the first and last IDs in the polygon that we are making are initialized but never changed once we expand the polygon.  Here we update the last ID because we added the way onto the end and below we do the opposite.